zerver/lib/bugdown/api_arguments_table_generator.py

```diff
@@ -48,26 +48,33 @@ def run(self, lines: List[str]) -> List[str]:
 
                     if not os.path.isabs(filename):
                         parent_dir = self.base_path
-                        filename = os.path.normpath(os.path.join(parent_dir, filename))
-
-                    try:
-                        if is_openapi_format:
-                            endpoint, method = doc_name.rsplit(':', 1)
-                            arguments = get_openapi_parameters(endpoint, method)
-                        else:
-                            with open(filename, 'r') as fp:
-                                json_obj = ujson.load(fp)
-                                arguments = json_obj[doc_name]
-
+                        filename = os.path.normpath(os.path.join(parent_dir,
+                                                                 filename))
+
+                    if is_openapi_format:
+                        endpoint, method = doc_name.rsplit(':', 1)
+
+                        try:
+                            arguments = get_openapi_parameters(endpoint,
+                                                               method)
+                        except KeyError as e:
+                            # Don't raise an exception if the "parameters"
+                            # field is missing; we assume that's because the
+                            # endpoint doesn't accept any parameters
+                            if e.args != ('parameters',):
+                                raise e
+
+                            arguments = []
```

Comment:
This is not a huge deal but could you please move this declaration to before the `try` statement? I think it is less confusing. :)